src/app.ts

```diff
@@ -172,6 +236,51 @@ function registerRoutes(app: Express): void {
   );
 }
 
+/**
+ * Initializes SpidStrategy for passport and setup /login route.
+ */
+function registerLoginRoute(
+  app: Express,
+  newSpidStrategy: passport.Strategy
+): void {
+  // Add the strategy to authenticate the proxy to SPID.
+  passport.use("spid", newSpidStrategy);
+  const spidAuth = passport.authenticate("spid", { session: false });
+  app.get("/login", spidAuth);
+
+  app.get("/metadata", (_0, res) => {
+    const metadata = ((newSpidStrategy as unknown) as SpidStrategy).generateServiceProviderMetadata(
+      samlCert()
+    );
+    res.type("application/xml").send(metadata);
+  });
+
+  app.post("/assertion-consumer-service", (req, res, next) => {
+    log.debug("dentro assertion-consumer-service");
+    passport.authenticate("spid", async (err, user) => {
```

Comment:
I don't think `authenticate` supports async handlers, can you check?

Reply:
Return type of `authenticate` is `any`, so an async function is correctly accepted and there's no problem in using it providing that the eventual exceptions are properly handled.
Anyway, the `async` keyword is not needed here, because at the current stage of implementation no `await` is used inside `authenticate`.